resources/prosody-plugins/mod_muc_census.lua

```diff
@@ -0,0 +1,71 @@
+-- report list of rooms with # of members and created date, filtered on test rooms
+
+local jid = require "util.jid";
+local json = require "util.json";
+local iterators = require "util.iterators";
+
+local have_async = pcall(require, "util.async");
+if not have_async then
+    module:log("error", "requires a version of Prosody with util.async");
+    return;
+end
+
+local async_handler_wrapper = module:require "util".async_handler_wrapper;
+
+local tostring = tostring;
+
+-- required parameter for custom muc component prefix, defaults to "conference"
+local muc_domain_prefix = module:get_option_string("muc_mapper_domain_prefix", "conference");
+
+--- handles request to get number of participants in all rooms
+-- @return GET response, containing json in this form:
+--
+--     {
+--         "room_census": [
+--             {
+--                 "room_name": "<muc name>",
+--                 "participants": <# participants>,
+--                 "created_time": <unix timestamp>,
+--             },
+--             ...
+--         ]
+--     }
+--
+function handle_get_room_census(event)
+    local host_session = prosody.hosts["conference." .. tostring(module.host)]
+    if not host_session or not host_session.modules.muc then
+        return { status_code = 400; }
+    end
+
+    room_data = {}
+    for room in host_session.modules.muc.each_room() do
+        if not tostring(room.jid):find("__jicofo-health-check", 1, true) then -- filter out test rooms
```

Comment:
@damencho is this the best way to filter out the rooms that don't have users? I also notice that they have `nil` as their created time, which could be another way to approach it

Reply:
Do it like this https://github.com/jitsi/jitsi-meet/blob/2008c9035944e2cc363bc47f72e07529c47289f8/resources/prosody-plugins/mod_muc_allowners.lua#L59

Reply:
changed, thanks